code/modules/assembly/flash.dm

```diff
@@ -88,6 +88,8 @@
 
 /obj/item/device/assembly/flash/proc/flash_carbon(mob/living/carbon/M, mob/user = null, power = 15, targeted = 1)
 	add_logs(user, M, "flashed", src)
+	if(M.disabilities & BLIND) //you can't flash a blind person!
+		return FALSE
 	if(user && targeted)
 		if(M.flash_act(1, 1))

```

Comment:
the check should be in here, though

Reply:
no, as I got stunned while blind with flash

Reply:
it should be in the flash_act proc.